### PR TITLE
Centcom Official ghost role

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

Centcom Official as a ghost role. Just a fluff role really. Good example for future ghost roles though.

Only has maint access.

**Changelog**

:cl: Swept
- add: Centcom Official ghost role

### CODE REVIEW

File: Resources/Prototypes/Entities/Effects/Markers/spawn_points.yml [hunk 1]
```diff
@@ -27,3 +26,13 @@
...
+- type: entity
+  name: centcom official spawn point
+  id: SpawnPointCentcomOfficial
+  parent: SpawnPoint
+  components:
+  - type: GhostRoleMobSpawnerComponent
+  - type: Sprite
+    state: Centcom
+  - type: SpawnPoint
+    job_id: SpawnPoint
```
Comment (Zumorica) [inline Resources/Prototypes/Entities/Effects/Markers/spawn_points.yml:38]: This is not how ghost roles work.
Ghost roles don't need nor use the SpawnPoint component.
This specific ghost role component takes the following variables:
- prototype: The prototype that will be used for the mob.
- deleteOnSpawn: whether this should be deleted after someone takes over the role.
- availableTakeovers: How many times this ghost role can be taken over. Defaults to 1, a new entity will be created per takeover.
- makeSentient: Whether the mob should be made sentient on spawn.
- name: The name for the ghost role on the ghost role window
- description: The description for the ghost role on the ghost role window

File: Resources/Maps/saltern.yml [hunk 2]
```diff
@@ -40729,4 +40729,10 @@ entities:
   - pos: -36.5,4.5
     parent: 853
     type: Transform
+- uid: 4724
+  type: SpawnPointCentcomOfficial
+  components:
+  - pos: -36.5,-6.5
+    parent: 853
+    type: Transform
```
Comment (Zumorica) [inline Resources/Maps/saltern.yml:40737]: This shouldn't really be part of the map unless you remove the ghost role component from it

**Review by Zumorica — CHANGES_REQUESTED** [review]
I see what you were trying to do, but ghost roles weren't meant to be used like this.
You will need to make a centcomm official mob prototype if you want to go with this using ghost roles. Although honestly, given how snowflakey human spawning is, might be best not to use ghost roles at all, or let admins on the live servers handle the creation of ghost roles (as they have done up until now)

**Comment by SweptWasTaken** [comment]
> 
> 
> I see what you were trying to do, but ghost roles weren't meant to be used like this.
> You will need to make a centcomm official mob prototype if you want to go with this using ghost roles. Although honestly, given how snowflakey human spawning is, might be best not to use ghost roles at all, or let admins on the live servers handle the creation of ghost roles (as they have done up until now)

Guess I misread the spec. I'll close this until it becomes possible without creating a separate mob prototype.

**Comment by SweptWasTaken** [comment]
Actually I remembered I made some changes to the RSIs so i'll just make this to spec anyway, easy to convert when the above is available.

**Comment by SweptWasTaken** [comment]
Ready to go, applied reviews and just made it a mob. However, spawned mobs are broken right now with UtilityAI so this is nonfunctional until @metalgearsloth fixes it. I'd still like it merged pretty please.

**Comment by SweptWasTaken** [comment]
Tests fixed

File: Resources/Prototypes/Entities/Mobs/Player/human.yml [hunk 3]
```diff
@@ -24,3 +24,20 @@
...
+    sprite: Interface/Misc/markers.rsi
+    state: Centcom
+  - type: GhostRoleMobSpawner
+    prototype: HumanMob_CentcomOfficial
+    deleteOnSpawn: true
+    makeSentient: false
+    name: centcom official
+    description: Inspect the station, jot down performance reviews for heads of staff, bug the Captain.
+  - type: UtilityAI
+    startingGear: CentcomGear
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/Player/human.yml:43]: Is this being used for the starting gear or should it wander.

Comment (SweptWasTaken) [reply Resources/Prototypes/Entities/Mobs/Player/human.yml:43]: starting gear

**Pull request merged**